src/core/UBPersistenceManager.cpp

```diff
@@ -1091,39 +1080,30 @@ void UBPersistenceManager::moveSceneToIndex(std::shared_ptr<UBDocumentProxy> pro
 
 std::shared_ptr<UBGraphicsScene> UBPersistenceManager::loadDocumentScene(std::shared_ptr<UBDocumentProxy> proxy, int sceneIndex, bool cacheNeighboringScenes)
 {
-    std::shared_ptr<UBGraphicsScene> scene = nullptr;
-
-    if (mSceneCache.contains(proxy, sceneIndex))
-    {
-        return mSceneCache.value(proxy, sceneIndex);
-    }
-    else
-    {
-        scene = UBSvgSubsetAdaptor::loadScene(proxy, sceneIndex);
-        if (scene)
-        {
-            mSceneCache.insert(proxy, sceneIndex, scene);
-        }
-        else
-        {
-            qWarning() << "could not load scene for document : " << proxy->persistencePath() << ", scene index : " << sceneIndex;
-            return nullptr;
-        }
-    }
+    mSceneCache.prepareLoading(proxy, sceneIndex);
+    auto scene = mSceneCache.value(proxy, sceneIndex);
+    qDebug() << "loadDocumentScene: got result from cache";
 
     if (cacheNeighboringScenes)
```

Comment:
Hi @letsfindaway, 

I'm going to change the cacheNeighboringScenes logic so it caches as much pages as authorized by the pageCacheSize setting, like following : 

```cpp
int lowerLimitNeighbor = std::max(0, sceneIndex - (UBSettings::settings()->pageCacheSize->get().toInt()/2));
for (int i = lowerLimitNeighbor; i< std::max(UBSettings::settings()->pageCacheSize->get().toInt() - lowerLimitNeighbor, proxy->pageCount()-1); i++)
{
    mSceneCache.prepareLoading(proxy,i);
}
```

Did you have some reasons not to do so ? Do you have any suggestions about this change ?

Also, I found, holding arrow keys to go to others pages where a video item can be found in some of them, that the application crashes (systematically) when the cache is full and that a  page containing a media has to be removed: 
```
cache full, size 20
removing page 27 of "OpenBoard Document 2024-09-27 10-51-26.623"
ASSERT failure in UBGraphicsVideoItem: "Called object is not of the correct type (class destructor may have already run)",
```

 Could you look at this ?

Reply:
> Did you have some reasons not to do so ? Do you have any suggestions about this change ?

No specific reason. One argument could however be that currently the `prepareLoading` does not maintain a queue, but starts loading all scenes in parallel. This causes that it takes longer until the first page was loaded. So it might be counterproductive to load all pages immediately.

> Also, I found, holding arrow keys to go to others pages where a video item can be found in some of them, that the application crashes (systematically) when the cache is full and that a page containing a media has to be removed:

Cannot reproduce this with a document where I have put an audio and a video on some of the pages. Can you provide me with a document where this happens?

But during testing I found that dragging the object to the thumbnail instead to the board causes an immediate crash.

Reply:
For the crash when dragging an object to the thumbnail view:
https://github.com/OpenBoard-org/OpenBoard/blob/fe2bbd3f8173eec555388dce792ca1b45258550c/src/gui/UBBoardThumbnailsView.cpp#L327-L340

Change the `else` in line 336 to `else if (mDropSource)` to avoid that DnD is executed without a valid drop source.

However I wonder why this `else` exists: can you drag any object from outside the thumbnails view to that view? I thought that the event source must always be the same widget as the drop target.

Reply:
> Can you provide me with a document where this happens?

Thanks for the documents. I tried it with all three of them. Still cannot reproduce the problem with the current `dev` branch. That's strange.

What I'm doing:

- Load the document from the Documents mode.
- Press and hold "arrow right" to flip through all pages.
- Same in the other direction, even multiple times.
- I also started the video on one of the pages and then moved away from that page. Also no crash.

Reply:
> Can you provide me with a document where this happens?

Done.

> No specific reason. One argument could however be that currently the `prepareLoading` does not maintain a queue, but starts loading all scenes in parallel. This causes that it takes longer until the first page was loaded. So it might be counterproductive to load all pages immediately.

Oh I indeed thought loading requests were queued and performed during idle time one at a time. So I thought it would not directly impact the user, just potentially indirectly because of the memory and cpu usage. 

Do you think anything is possible with slight changes or will it take more important changes than what I'm doing ?

Reply:
> > Can you provide me with a document where this happens?
> 
> Thanks for the documents. I tried it with all three of them. Still cannot reproduce the problem with the current `dev` branch. That's strange.
> 
> What I'm doing:
> 
> * Load the document from the Documents mode.
> * Press and hold "arrow right" to flip through all pages.
> * Same in the other direction, even multiple times.
> * I also started the video on one of the pages and then moved away from that page. Also no crash.

Ok so it's probably Qt related. Happening with the Qt 6.7.2 version

Reply:
Also happening with Qt 6.4.2 on a Debian virtual machine. Did you try with a Qt 5 based openboard ?

Reply:
Ok, confirmed and it is an issue with Qt 6. The reason is here:
https://github.com/OpenBoard-org/OpenBoard/blob/fe2bbd3f8173eec555388dce792ca1b45258550c/src/domain/UBGraphicsMediaItem.cpp#L170-L201

The connections are different between Qt5 and Qt6. When deleting a media object, the destructor calls `mMediaObject->stop()`. This triggers the `playbackStateChanged` signal which is - when using Qt 6 - connected to `UBGraphicsVideoItem::mediaStateChanged`. But at this stage the destructor of `UBGraphicsVideoItem` has already run, so the object is no longer an instance of that class. 

The connection is still alive, even if it was created in a subclass. This is because connections are released in the `QObject` destructor, which runs later. In fact this might also be a problem for Qt5, but it seems not to be fatal as in Qt6.

I will work on a solution later this weekend, most probably by disconnecting these signals in the destructor.